crypto/src/action/output.rs

```diff
@@ -1,26 +1,80 @@
 use ark_ff::UniformRand;
+use bytes::Bytes;
 use rand_core::{CryptoRng, RngCore};
+use std::convert::{TryFrom, TryInto};
 
-use crate::{addresses::PaymentAddress, ka, memo::MemoPlaintext, note, value, Fq, Fr, Note, Value};
+use penumbra_proto::{transaction, Protobuf};
 
-pub const OVK_WRAPPED_LEN_BYTES: usize = 80;
+use crate::{
+    action::{
+        constants::{NOTE_ENCRYPTION_BYTES, OVK_WRAPPED_LEN_BYTES},
+        error::Error,
+    },
```

Comment:
nit: I think it's slightly preferable to import stuff in the same module with (at most one level of) `super`, rather from the crate root (i.e., relative to where we are, rather than relative to the crate), since it means that if the module tree gets moved around, it's less likely that the imports will need changing.